aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -510,4 +534,54 @@ Tensor& capped_random_cuda_(Tensor& self, int64_t to, Generator* gen) {
   return clamped_random_cuda_(self, 0, to, gen);
 }
 
+Tensor& normal_cuda_(Tensor& self, double mean, double std, Generator* gen) {
+  TORCH_CHECK(std > 0.0, "normal_ expects std > 0.0, but found std=", std);
+  auto iter = TensorIterator::nullary_op(self);
+  normal_kernel_cuda(*iter, mean, std, gen);
+  return self;
+}
+
+Tensor& normal_out_cuda(Tensor& output, const Tensor& mean_, double std, Generator* gen) {
+  auto mean = std::get<0>(expand_inplace(output, mean_.to(kCUDA)));
+  normal_cuda_(output, 0, std, gen);
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(output.scalar_type(), "normal_out_cuda_cadd", [&] {
+    at::add_out(output, output, mean, static_cast<scalar_t>(1));
```

Comment:
nit: If possible, it's better to use the in-place versions of these functions. I don't think it matters here, but the reason to prefer an actual in-place is that it will guarantee for you that aliasing is handled correctly (in principle, someone can incorrectly implement an `_out` kernel so that it doesn't work if the output aliases with inputs.)

Reply:
Updated with in-place add_